### DISCUSSION TITLE
Change markdown lexer colors in Textadept 12

### DISCUSSION THREAD

**Discussion by a1ess** [orbitalquark/textadept#410] [General]
Hi,

I'm testing version12 alpha 3 of Textadept on my Linux system. 

In v11.4 I had this lines in my `~/.textadept/themes/light.lua` to change the color and size of the headers (h1 to h4) of markdown files:

```lua

-- Markdown

styles.h1 = {fore = colors.red, size = fontsize+8}

styles.h2 = {fore = colors.red, size = fontsize+6}

styles.h3 = {fore = colors.red, size = fontsize+4}

styles.h4 = {fore = colors.red, size = fontsize+2}

```

(fontsize is a local variable where I define a base font size).

They work fine in 11.4, but do nothing in v. 12 (headers all the same size and magenta or purple, cannot tell which color it is).

Where can I find more information on how to change lexer colors and style in Textadept 12?

**Comment by orbitalquark** [comment]
Take a look at https://orbitalquark.github.io/textadept/manual.html#themes. It should help.

**Reply by a1ess** [reply]
Sorry, I posted by mistake before finishing writing. I had already looked at the themes and migration sections of the manual, but couldn't find the solution. I've updated the original post.

**Reply by orbitalquark** [reply]
Ah I see, sorry about it. It's not very clear, but if you navigate to a markdown header and use the "Tools > Show Style" menu, you'll see that the style is "heading_h1", so you would use the following:

    styles.heading_h1 = styles[lexer.HEADING] .. {size = size + 8}

I should probably add some documentation for that.

**Comment by a1ess** [comment]
It works. Thank you! 

Yes, I think expanding on the syntax in the Themes section of the manual  would be helpful. I don't think I could have figured out the new syntax by myself.

I also noticed that the syntax you indicated work for the elements that use a predefined tag (like headings) in the lexer. Other elements that do not use predefined tags, like the `hr` element in markdown, can be modified for example with:

```lua
styles.hr = {bold = true}
```

**Reply by orbitalquark** [reply]
I've updated the API documentation to mention the "Tools > Show Style" menu item for obtaining the name of the style under the caret for styles like "heading_h1": https://orbitalquark.github.io/textadept/api.html#view.styles

I didn't feel like the manual was the place for it since it's just a high level overview of using themes. It links to the API docs for the nitty-gritty details.
